cortex/built_ins/networks/utils.py

```diff
@@ -75,7 +75,13 @@ def apply_nonlinearity(x, nonlinearity, **nonlinearity_args):
         if isinstance(nonlinearity, str):
             nonlinearity = get_nonlinearity(nonlinearity)
         if callable(nonlinearity):
-            x = nonlinearity(x, **nonlinearity_args)
+            if isinstance(nonlinearity, nn.PReLU):
```

Comment:
Shouldn't Pytorch be handling this? But if there is an error, shouldn't we check whether `x` is on `cuda` ?

Reply:
Yes, changed to use `x.device`